editoast/editoast_derive/src/modelv2/args.rs

```diff
@@ -30,6 +33,20 @@ pub(super) struct ModelArgs {
     pub(super) data: ast::Data<util::Ignored, ModelFieldArgs>,
 }
 
+#[derive(Debug, PartialEq, Eq, Derivative, FromMeta)]
+#[derivative(Default)]
+pub(super) struct ImplPlan {
+    #[darling(default)]
+    #[derivative(Default(value = "Crud::ALL"))]
+    pub(super) ops: Crud,
+    #[darling(default)]
+    #[derivative(Default(value = "Crud::ALL"))]
+    pub(super) batch_ops: Crud,
```

Comment:
Shouldn't the default be some kind of `Crud:NONE` ? From the usage on the models, it feels like you specify `ops = crud` quite often, but not `batch_ops = crud` (often `batch_ops` isn't mentioned)

Reply:
This comment is also true for `list`

Reply:
There are two different "default values" involved here.

1. The ones specified in `derivative` are used when there's no `ImplPlan` to parse, i.e. without `#[model(gen(...))]`. In this case, we implement all traits available (how it worked before this feature), hence the `Crud::ALL`.
2. When there's a `#[model(gen(...))]` annotation but some fields are omitted however, the `#[darling(default)]` applies. For example, `#[model(gen(ops = crud))]` will generate an `ImplPlan { ops: Crud { <crud>: true}, batch_ops: Crud { <crud>: false }, list: false }` as the `Default` of `Crud` sets everything to `false` (cf. `#[derive(Default)] struct Crud`).

Reply:
Wow okay, that was a skill issue on my part
That's quite convoluted

Reply:
It feels quite counterintuitive that the less you write in `gen` the less code is added, but not writing `gen` at all writes the most code

Reply:
I see what you mean, but if not writing `gen` at all generates no code, you end up in a situation where `derive(ModelV2)` works without `gen` configuration but doesn't generate anything useful. It'd be like if `#[derive(Deserialize)]` doesn't make your type deserializable unless you add an extra `#[serde(please_do_gen_deserialize)]`.

Maybe we should make `gen` required to eliminate this behavior. Or maybe `gen` isn't the best naming for that. Not sure what to do.

Reply:
I've also added https://github.com/OpenRailAssociation/osrd/pull/7898#discussion_r1742236086 in a comment as the behavior isn't exactly straightforward.

Reply:
Actually, making `gen` mandatory isn't so bad. Besides the fact that it's more intiutive in terms of code generation (identified by @Castavo ). It forces the API user to ask what he really needs.
We could check if `ops` and `ops_batch` are both empty and throw a compilation error with a meaningful message.

Reply:
Done. I'll let you resolve if you're ok with the change. Auto-merge is on.